### ISSUE TITLE
Improve this line, if username and passwords are allowed to be ANY characters

### ISSUE THREAD

**Issue by mastershin** [neiljones12/VoxelCloud-Python#1]
https://github.com/neiljones12/VoxelCloud-Python/blob/8f9c84c1925b9d96dd74fc9ab3bad9c55165935b/API/login.py#L25-L27

**Comment by neiljones12** [comment]
Added an additional check to verify the lengths of the input

![1](https://user-images.githubusercontent.com/1053868/42065256-c244ebaa-7aee-11e8-9401-42ead2f64e83.PNG)

**Comment by mastershin** [comment]
If MAX_LENGTH is defined to be 50, how would you improve the code?

**Comment by mastershin** [comment]
Also password is allowed have a special characters and spaces.

**Comment by neiljones12** [comment]
Updated the password validation, is this ok ? 

![1](https://user-images.githubusercontent.com/1053868/42073394-40dff08c-7b1b-11e8-8ad9-89fc01d6640e.PNG)

**Comment by mastershin** [comment]
Actually, remove regex validation completely.  I was looking for alternate solution to improve your SQL statement.

**Comment by neiljones12** [comment]
Password verification without regex
I am iterating through the string and checking to see if our conditions are satisfied. 

![capture](https://user-images.githubusercontent.com/1053868/42076317-817039f0-7b29-11e8-8fc1-acfabe715c45.PNG)

**Comment by mastershin** [comment]
I meant to completely get rid of password validation, since psycopg will handle it.  No need to validate.  password can also contain Unicode (international characters).

**Comment by neiljones12** [comment]
Instead of saving the password as plain text, i am saving them as hashed passwords.
I updated the login function to check the input password against the hashed password from the database.

Is this the correct approach ?


![pass2](https://user-images.githubusercontent.com/1053868/42106603-75d31472-7b89-11e8-81a7-4e6eef9ae664.PNG)

**Comment by mastershin** [comment]
Yes, you're on the correct path.  We should never store plain password to the database.  But, I'm curious how you would improve sql statement then.

**Comment by neiljones12** [comment]
I changed the query by not passing the password in it. I am first checking to see if the user exists, if yes, i'm checking the input password against the hashed password.

Is this the correct approach ?

![password](https://user-images.githubusercontent.com/1053868/42110540-53411064-7b96-11e8-93c0-0c4c279aa814.PNG)

**Comment by mastershin** [comment]
Yes, much better approach.  You may want to rename the db field to
PasswordHash.  By the way, the customer may login with email address or
some assigned username, so, treat it as a string.

**Comment by neiljones12** [comment]
I am treating the first parameter as a string now and changed the Password column to PasswordHash in the database

![capture](https://user-images.githubusercontent.com/1053868/42111475-74b86f8c-7b99-11e8-8635-f72273d55947.PNG)
